ludwig/datasets/mixins/load.py

```diff
@@ -38,13 +38,20 @@ def load_processed_dataset(self, split) -> Union[pd.DataFrame,
         :param split: Splits along 'split' column if present
         :returns: A pandas dataframe
         """
-        dataset_csv = os.path.join(self.processed_dataset_path, self.csv_filename)
+        dataset_csv = os.path.join(self.processed_dataset_path,
+                                   self.csv_filename)
         data_df = pd.read_csv(dataset_csv)
+        if SPLIT in data_df:
+            data_df[SPLIT] = pd.to_numeric(data_df[SPLIT])
         if split:
-            training_set = data_df[data_df[SPLIT] == "0"]
-            val_set = data_df[data_df[SPLIT] == "1"]
-            test_set = data_df[data_df[SPLIT] == "2"]
-            return training_set, test_set, val_set
+            if SPLIT in data_df:
+                training_set = data_df[data_df[SPLIT] == 0]
+                val_set = data_df[data_df[SPLIT] == 1]
+                test_set = data_df[data_df[SPLIT] == 2]
+                return training_set, test_set, val_set
+            else:
+                raise ValueError("The dataset does not have splits, "
```

Comment:
if we're here whats the ask from the user, do they have to re-create the dataset?  I am wondering if we should have some explicit instructions on what they need to do?

Reply:
Good question.  In this case, they need to set `split=False` to return the entire dataset, then either manually split themselves or let Ludwig automatically assign splits (this will happen automatically if the user passes in `mode.train(dataset=dataset)`).

Reply:
Thought about it and decided for the ValueError for the following reasons:

- users will call `train, test, val = load(split=True)` so returning a single dataset would anyway raise an error, so better be explicit about it.
- the datasets that we are providing here either have official splits or they don't. In case they don't I'd rather have either the user come up with a split of having them trusting Ludwig with the splits, so that the dataset interface task is just to provide datasets as they are, without any surprise.